c-bindings/wrapper.h

```diff
@@ -0,0 +1,13 @@
+#include "picosha2.hpp"
+#include "../src/verifier.hpp"
+
+extern "C" {
+    typedef struct {
+        uint8_t* data;
+        size_t length;
+    } ByteArray;
```

Comment:
with my suggestion you wouldn't need this type. I think it's fine to have a C interface where the caller must know about the (fixed) size of buffers. You immediately wrap it in a rust binding where this information is encoded in the type anyway.